elasticsearch/elasticsearch.d.ts

```diff
@@ -0,0 +1,253 @@
+        id: string;
+        body: T;
+        consistensy?: string;
+        parent?: string;
+        replication?: string;
+        routing?: string;
+        timeout?: Date | number;
+        timestamp?: Date | number;
+        version?: number;
+        versionType?: string;
+    }
+
+    export interface ScrollParams extends GenericParams {
+        scroll: string;
+        scrollId: string;
+    }
+
+    export interface SearchParams extends GenericParams {
+        index?: string;
+        type?: string | string[] | number;
+        body?: any;
+        q?: string;
+        scroll?: string;
+        search_type?: string;
+        fields?: string[];
+        size?: number;
+        sort?: string | string[] | boolean;
+        _source?: string | string[] | boolean;
+        _sourceExclude?: string | string[] | boolean;
+        _sourceInclude?: string | string[] | boolean;
+        stats?: string | string[] | boolean;
+        suggestField?: string;
+        suggestSize?: number;
+        suggestText?: string;
+        timeout?: Date | number;
+    }
+
+    export interface MSearchParams extends GenericParams {
+        index?: string | string[] | Boolean;
+        type?: string | string[] | Boolean;
+        search_type?: string;
+    }
+
+    export interface MGetParams extends GenericParams {
+        fields?: string | string[] | Boolean;
+        preference?: string;
+        realtime?: Boolean;
+        refresh?: Boolean;
+        _sourceExclude?: string | string[] | boolean;
+        _sourceInclude?: string | string[] | boolean;
+        index?: string;
+        type?: string;
+    }
+
+    export interface IndicesIndexExitsParams extends GenericParams {
+        index: string | string[] | boolean;
+        ignoreUnavailable?: boolean;
+    }
+
+    export interface PingParams extends GenericParams {
+        requestTimeout?: number;
+        hello?: string;
+    }
+
+    export interface DeleteDocumentParams extends GenericParams {
+        index: string;
+        type: string;
+        id: string;
+        refresh?: boolean;
+    }
+
+    export interface UpdateDocumentParams extends GenericParams {
+        index: string;
+        type: string;
+        id: string;
+        body?: string | any;
+        version?: Number;
+        timesstamp?: Date | Number;
+        scriptedUpsert?: Boolean;
+        scriptId?: any;
+        script?: any;
+        routing?: string;
+        retryOnConflict?: Number;
+        refresh?: Boolean;
+        parent?: string;
+        lang?:string;
+        fields?: string | string[] | Boolean;
+        consistensy?: string;
+    }
+
+    export interface SuggestParams extends GenericParams {
+        ignoreUnavailable?: boolean;
+        allowNoIndices?: boolean;
+        expandWildcards?: string;
+        preference?: string;
+        routing?: string;
+        source?: string;
+        body: string | any;
+        index: string | string[] | boolean;
+    }
+
+    export interface IPromise<T> {
```

Comment:
you can refer to `PromiseLike<T>` defined in lib.d.ts instead of defining your own promise.